js/src/util/args.ts

```diff
@@ -132,14 +132,14 @@ function _selectFieldArgs<T extends { [key: string]: DataType }>(vals: any[], re
     while (++idx < len) {
         val = vals[idx];
         if (val instanceof Column && (values[++valueIndex] = val)) {
-            fields[++fieldIndex] = val.field.clone(keys[idx], val.type, val.nullCount > 0);
+            fields[++fieldIndex] = val.field.clone(keys[idx], val.type, true);
         } else {
             ({ [idx]: field = idx } = keys);
             if (val instanceof DataType && (values[++valueIndex] = val)) {
-                fields[++fieldIndex] = Field.new(field, val as DataType) as Field<T[keyof T]>;
+                fields[++fieldIndex] = Field.new(field, val as DataType, true) as Field<T[keyof T]>;
             } else if (val && val.type && (values[++valueIndex] = val)) {
                 val instanceof Data && (values[valueIndex] = val = Vector.new(val) as Vector);
-                fields[++fieldIndex] = Field.new(field, val.type, val.nullCount > 0) as Field<T[keyof T]>;
+                fields[++fieldIndex] = Field.new(field, val.type, true) as Field<T[keyof T]>;
```

Comment:
Why are all these checks hard-coded to true now?

Reply:
@TheNeuralBit Ah yes, I forgot I had to change this. It's not for testing, but for streaming. The problem goes like this:

* We start pulling from a source stream and appending values to the builder
* The builder reaches the high-water-mark, and yields a RecordBatch to the Writer
* The Writer sees it hasn't written a Schema yet, so it uses the Schema of the first batch

If the first RecordBatch doesn't contain nulls, the Schema Message's Field will have `nullable=false`. If the source later yields a null, changing the Field's `nullable=true` won't be reflected in the stream since the Schema has already been sent. It's hacky, but I changed the default value of `nullable=true` to accommodate this unless explicitly set.

Reply:
Technically `nullable` doesn't indicate the existence of nulls, just that it's possible there will be nulls. I'm not sure if other implementations infer `nullable=true` to also mean there's a non-zero-length null bitmap buffer. If so, we can update the writer to synthesize one for compatibility.

Reply:
Ah I see. The spec says you're allowed to elide validity bitmaps when the null count is 0, so I think we're ok there:

> Arrays having a 0 null count may choose to not allocate the null bitmap. Implementations may choose to always allocate one anyway as a matter of convenience, but this should be noted when memory is being shared.

(from http://arrow.apache.org/docs/format/Layout.html#null-bitmaps)

Doesn't need to happen here, but could we set nullable based on the builder options? e.g. it there are no `null_values`. I guess we would probably need another builder option (like `nullable = false`) to do that right, since you could still append `null` or `undefined` even without `null_values`.